src/main/java/org/commcare/util/EncryptionUtils.java

```diff
@@ -53,33 +121,111 @@ public static String encrypt(String message, String key) throws EncryptionExcept
         }
     }
 
-
-    private static SecretKey getSecretKeySpec(String key) throws EncryptionException {
-        final int KEY_LENGTH_BIT = 256;
+    /**
+     * Converts a Base64 encoded key into a SecretKey, PrivateKey or PublicKey, depending on the
+     * algorithm and cryptographic operation
+     *
+     * @param algorithm              the algorithm to be used to encrypt/decrypt
+     * @param base64encodedKey       key in String format
+     * @param cryptographicOperation Cryptographic operation where the key is to be used, relevant
+     *                               to the RSA algorithm
+     * @return Secret key, Public key or Private Key to be used
+     */
+    private static EncryptionKeyAndTransformation getKey(String algorithm,
```

Comment:
> EncryptionKeyProvider is used to support Key Store operations

I don't think that's right interpretation when we have a `BasicEncryptionKeyProvider` which doesn't even have a Keystore. I think we should interpret `EncryptionKeyProvider` instead as a class that provides encryption key (it can be from keystore for `KeyStoreEncryptionKeyProvider` (rename `EncryptionKeyProvider` to this for clarity) , or can be formed manually like this in the `BasicEcryptionKeyProvider` )

Reply:
Another aspect here is that `BasicEncryptionKeyProvider` is more of a dummy implementation, just to satisfy test requirements. Moving `getKey` there would require also making it also available to `EncryptionKeyProvider`, unless we define a dependency between the two.

Reply:
> BasicEncryptionKeyProvider is more of a dummy implementation, just to satisfy test requirements

 XpathE/DncryptFunc use `BasicEncryptionKeyProvider` right ? 
 
 >  Moving getKey there would require also making it also available to EncryptionKeyProvider
 
 I agree, although I would replace the existing `retrieveKeyFromKeyStore` with `getKey` implementation. This would simplify the code overall as you won't need to have separate methods like `encryptWithKeyStore` and `encryptWithBase64EncodedKey` but can simply rely on `EncryptionKeyProvider` to supply the right key.